sys/trickle/trickle.c

```diff
@@ -30,63 +27,61 @@ void trickle_callback(trickle_t *trickle)
     if ((trickle->c < trickle->k) || (trickle->k == 0)) {
         (*trickle->callback.func)(trickle->callback.args);
     }
+
+    trickle_interval(trickle);
 }
 
 void trickle_interval(trickle_t *trickle)
 {
-    uint32_t max_interval;
+    uint32_t old_interval = trickle->I;
```

Comment:
`trickle->I` is probably already changed to`trickle->I = max_interval;` in [L43](https://github.com/RIOT-OS/RIOT/pull/7161/files#diff-c5efcfdb3eedbfa8c2b441ecda645879R43).
So we would use the "wrong" value to compute `trickle->t` :)